lib/api/CAnomalyJob.cc

```diff
@@ -1574,8 +1581,12 @@ CAnomalyJob::detectorForKey(bool isRestoring,
     return itr->second;
 }
 
-void CAnomalyJob::pruneAllModels() {
-    LOG_INFO(<< "Pruning all models");
+void CAnomalyJob::pruneAllModels(std::size_t buckets) {
+    if (buckets == 0) {
+        LOG_INFO(<< "Pruning all models");
```

Comment:
This message is now out-of-sync with the one two lines below, as the meaning of "all" has been implicitly changed.  According to the method that gets called it "prunes ancient models that are effectively dead".  We probably shouldn't put that in a log message, but maybe "Pruning obsolete models"?  If people see this message and the one two lines below then we need to make clear that this is generally less aggressive than the other type of pruning.

There's also an implication for the docs: we need to make clear that models get pruned when we expect them to contain no information at all _even when pruning is not enabled in the config_.